### ISSUE TITLE
CreatePersistentSubscriptionAsync fails against a 20.6.0 follower when using the 5.0.9 dotnet client

### ISSUE THREAD

**Issue by 01100010011001010110010101110000** [EventStore/EventStore#2646]
**Describe the bug**
When executing `CreatePersistentSubscriptionAsync` from the `5.0.9` dotnet client against a Follower 20.6.0 node, the following exception is raised

```
System.ArgumentNullException: Value cannot be null.
Parameter name: ipString
   at System.Net.IPAddress.Parse(String ipString)
   at EventStore.ClientAPI.Messages.ClientMessage.NotHandled.MasterInfo.get_ExternalTcpEndPoint()
   at EventStore.ClientAPI.ClientOperations.OperationBase`2.InspectNotHandled(TcpPackage package)
   at EventStore.ClientAPI.ClientOperations.OperationBase`2.InspectPackage(TcpPackage package)
--- End of stack trace from previous location where exception was thrown ---
   at EventStore.ClientAPI.Internal.EventStoreNodeConnection.CreatePersistentSubscriptionAsync(String stream, String groupName, PersistentSubscriptionSettings settings, UserCredentials credentials)
```

**To Reproduce**
1. Execute `CreatePersistentSubscriptionAsync` against a Follower node in a 20.6.0 cluster using the `5.0.9` client

**Expected behavior**
The client should successfully terminate the connection to the Follower node, connect to master, and create the persistent subscription. This does work from the `5.0.9` client against a slave node in a v5 cluster, or from the `20.6.0` client against a `20.6.0` cluster. 

**Actual behavior**
```
System.ArgumentNullException: Value cannot be null.
Parameter name: ipString
   at System.Net.IPAddress.Parse(String ipString)
   at EventStore.ClientAPI.Messages.ClientMessage.NotHandled.MasterInfo.get_ExternalTcpEndPoint()
   at EventStore.ClientAPI.ClientOperations.OperationBase`2.InspectNotHandled(TcpPackage package)
   at EventStore.ClientAPI.ClientOperations.OperationBase`2.InspectPackage(TcpPackage package)
--- End of stack trace from previous location where exception was thrown ---
   at EventStore.ClientAPI.Internal.EventStoreNodeConnection.CreatePersistentSubscriptionAsync(String stream, String groupName, PersistentSubscriptionSettings settings, UserCredentials credentials)
```

**Config/Logs/Screenshots**
Three node cluster, I can provide logs or a trivial project to repro the problem if that would help. 

**EventStore details**
 - EventStore server version: `20.6.0`

 - Operating system: Ubuntu 18.04

 - EventStore client version (if applicable): `5.0.9`

**Comment by hayley-jean** [comment]
Thanks for reporting this @01100010011001010110010101110000, could you please provide us with the configuration you are using for the client and the server, as well as with some reproduction code if possible?

**Comment by 01100010011001010110010101110000** [comment]
Certainly, [here's](https://gist.github.com/01100010011001010110010101110000/072bf00e0679c712c760695781c89cf4) a gist with the program.cs and eventstore.conf

**Comment by 01100010011001010110010101110000** [comment]
I seem to be encountering a similar error when calling `ConnectToPersistentSubscription`. The setup is the same, but the stack trace is below. Seems `ExternalHttpIp` is nil on the member info object

```
 Discovering attempt 1/1 failed with error: System.ArgumentNullException: Value cannot be null. (Parameter 'ipString')
   at System.Net.IPAddress.Parse(String ipString)
   at EventStore.ClientAPI.Internal.ClusterDnsEndPointDiscoverer.ArrangeGossipCandidates(MemberInfoDto[] members)
   at EventStore.ClientAPI.Internal.ClusterDnsEndPointDiscoverer.GetGossipCandidatesFromOldGossip(IEnumerable`1 oldGossip, IPEndPoint failedTcpEndPoint)
   at EventStore.ClientAPI.Internal.ClusterDnsEndPointDiscoverer.DiscoverEndPoint(IPEndPoint failedEndPoint)
   at EventStore.ClientAPI.Internal.ClusterDnsEndPointDiscoverer.<>c__DisplayClass10_0.<DiscoverAsync>b__0().
```

**Comment by 01100010011001010110010101110000** [comment]
It seems the `20.6.0` gossip returns `httpEndPointIp` instead of the old `InternalHttpIp` and `ExternalHttpIp`, so those attributes are both `null` when the gossip is deserialized

**Comment by 01100010011001010110010101110000** [comment]
This should be fixed after those two PRs (I've been running a patched v5 client build from the `5.0.9` tag plus those two branches) so I'm closing this out
